pkg/lib/client/openstack/client.go

```diff
@@ -895,42 +876,38 @@ func (c *Client) volumeAPI(object interface{}, opts interface{}) (err error) {
 	switch object.(type) {
 	case *[]Volume:
 		object := object.(*[]Volume)
-		switch opts.(type) {
+		switch opts := opts.(type) {
 		case *VolumeListOpts:
-			opts := opts.(*VolumeListOpts)
 			err = c.volumeList(object, opts)
 		default:
 			err = c.unsupportedTypeError(object)
 		}
 	case *Volume:
 		object := object.(*Volume)
 		var volume *volumes.Volume
-		switch opts.(type) {
+		switch opts := opts.(type) {
 		case *GetOpts:
-			opts := opts.(*GetOpts)
 			ID := opts.ID
 			volume, err = volumes.Get(c.blockStorageService, ID).Extract()
 			if err != nil {
 				return
 			}
 			*object = Volume{Volume: *volume}
 		case *VolumeCreateOpts:
-			opts := opts.(*VolumeCreateOpts)
 			volume, err = volumes.Create(c.blockStorageService, opts).Extract()
 			if err != nil {
 				return
 			}
 			*object = Volume{Volume: *volume}
 		case *VolumeUpdateOpts:
-			opts := opts.(*VolumeUpdateOpts)
 			volume, err = volumes.Update(c.blockStorageService, object.ID, opts).Extract()
 			if err != nil {
 				return
 			}
 			*object = Volume{Volume: *volume}
 		case *DeleteOpts:
-			opts := volumes.DeleteOpts{Cascade: true}
-			err = volumes.Delete(c.blockStorageService, object.ID, opts).ExtractErr()
+			deleteOpts := volumes.DeleteOpts{Cascade: true}
```

Comment:
it was that way before so it's fine to keep it this way in this PR but wouldn't it make sense to rather set `Cascade` on `opts`?